src/panels/lovelace/cards/hui-button-card.js

```diff
@@ -0,0 +1,122 @@
+    <style>
+      ha-icon {
+        display: flex;
+        margin: auto;
+      }
+      paper-button {
+        display: flex;
+        margin: auto;
+        text-align: center;
+      }
+      .not-found {
+        flex: 1;
+        background-color: yellow;
+        padding: 8px;
+      }
+    </style>
+    <ha-card on-click='_handleClick'>
+      <paper-button>
+        <div id='Button'>
+          <ha-icon id='ButtonIcon' style='color: [[_computeColor(_stateObj)]]; width:40%; height:40%;"' icon='[[_computeIcon(_stateObj)]]'></ha-icon>
+          <span id='ButtonName'>[[_computeName(_stateObj)]]</span>
+          <span id='ButtonState'>[[_computeState(_stateObj)]]</span>
+        </div>
+      </paper-button>
+    </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object
+      },
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)'
+      },
+    };
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity || (config.tap_action === 'call-service' && !config.service)) throw new Error('Invalid card configuration');
+    this._config = Object.assign({ color: 'var(--primary-text-color)', show_state: true }, config);
+  }
+
+  _computeStateObj(states, entityId) {
+    return states && entityId in states ? states[entityId] : null;
+  }
+
+  _computeName(stateObj) {
+    const config = this._config;
+    if (!stateObj) {
+      this.$.Button.className = 'not-found';
+      return 'Entity not available: ' + this._config.entity;
+    }
+    return config.name ? config.name : stateObj.attributes.friendly_name;
+  }
+
+  _computeIcon(stateObj) {
+    const config = this._config;
+    if (!stateObj) return '';
+    return config.icon ? config.icon : stateIcon(stateObj);
+  }
+
+  _computeColor(stateObj) {
+    const config = this._config;
+    if (!stateObj) return '';
+    const color = stateObj.state === 'off' ? 'var(--disabled-text-color)' : config.color;
+    return stateObj.attributes.rgb_color ? `rgb(${stateObj.attributes.rgb_color.join(',')})` : color;
+  }
+
+  _computeState(stateObj) {
+    const config = this._config;
+    if (!stateObj) return '';
+    return config.show_state ? stateObj.state : '';
+  }
+
+  _handleClick() {
+    if (!this._stateObj) return;
+    const config = this._config;
+    const stateObj = this._stateObj;
+    const entityId = stateObj.entity_id;
+    switch (config.tap_action) {
+      case 'more-info':
+        this.fire('hass-more-info', { entityId });
+        break;
+      case 'call-service': {
+        const [domain, service] = config.service.split('.', 2);
+        const serviceData = Object.assign(
+          {}, { entity_id: entityId },
+          config.service_data
+        );
+        this.hass.callService(domain, service, serviceData);
+        break;
+      }
+      default:
+        toggleEntity(this.hass, entityId);
```

Comment:
I think that this default is a bit weird, it only works if we have an entity id.  What about scripts and services?

Reply:
If action is true I would set the tap_action to call-service